src/OpenColorIO/transforms/builtins/ACES.cpp

```diff
@@ -497,6 +497,73 @@ void Generate_roll_white_d65_ops(OpRcPtrVec & ops)
 
 }  // namespace ACES_OUTPUT
 
+namespace ACES2_OUTPUT
+{
+    void Generate_output_transform(
+        OpRcPtrVec & ops,
+        float peak_luminance,
+        const Primaries & limiting_pri,
+        const Primaries & encoding_pri,
+        float linear_scale,
+        bool scale_white)
+    {
+        // Clamp to AP1
+        MatrixOpData::MatrixArrayPtr matrixToAP1
+            = build_conversion_matrix(ACES_AP0::primaries, ACES_AP1::primaries, ADAPTATION_NONE);
+        CreateMatrixOp(ops, matrixToAP1, TRANSFORM_DIR_FORWARD);
+
+        const float upperBound = 8.f * (128.f + 768.f * (std::log(peak_luminance / 100.f) / log(10000.f / 100.f)));
+        CreateRangeOp(ops,
+            0., upperBound,
+            0., upperBound,
+            TRANSFORM_DIR_FORWARD);
+
+        CreateMatrixOp(ops, matrixToAP1, TRANSFORM_DIR_INVERSE);
+
+        // Display rendering
+        CreateFixedFunctionOp(ops, FixedFunctionOpData::ACES_OUTPUT_TRANSFORM_20_FWD, {
+            peak_luminance,
+            limiting_pri.m_red.m_xy[0], limiting_pri.m_red.m_xy[1],
+            limiting_pri.m_grn.m_xy[0], limiting_pri.m_grn.m_xy[1],
+            limiting_pri.m_blu.m_xy[0], limiting_pri.m_blu.m_xy[1],
+            limiting_pri.m_wht.m_xy[0], limiting_pri.m_wht.m_xy[1],
+        });
+
+        // Post transform clamp
+        const double normPeakLuminance = peak_luminance / 100.f;
+        CreateRangeOp(ops,
+            0., normPeakLuminance,
+            0., normPeakLuminance,
+            TRANSFORM_DIR_FORWARD);
+
+        // White point simulation
+        if (scale_white)
+        {
+            MatrixOpData::MatrixArrayPtr matrixLimToOut
+                = build_conversion_matrix(limiting_pri, encoding_pri, ADAPTATION_NONE);
+
+            MatrixOpData::Offsets white(1.f, 1.f, 1.f, 0.f);
+            white = matrixLimToOut->inner(white);
+
+            const double scale = 1. / std::max(std::max(white[0], white[1]), white[2]);
+            const double scale4[4] = { scale, scale, scale, 1. };
+            CreateScaleOp(ops, scale4, TRANSFORM_DIR_FORWARD);
+        }
+
+        // Linear scale factor
+        if (linear_scale != 1.f)
+        {
+            const double scale = linear_scale;
+            const double scale4[4] = { scale, scale, scale, 1. };
+            CreateScaleOp(ops, scale4, TRANSFORM_DIR_FORWARD);
+        }
+
+        MatrixOpData::MatrixArrayPtr matrixToXYZ
+            = build_conversion_matrix_to_XYZ_D65(limiting_pri, ADAPTATION_NONE);
+        CreateMatrixOp(ops, matrixToXYZ, TRANSFORM_DIR_FORWARD);
```

Comment:
Although the limiting_pri is sometimes D60, it does seem that the ADAPTATION_NONE is the correct option to use to match what the CTL is doing for the current set of output transforms. But I am puzzling over it.

All of the current output transforms fall in two sets: either D65, or D60 in a D65 container. Unlike ACES 1, aside from the DCDM, there are no displays that are not calibrated to D65. However, if there was a desire to add a D60 calibrated display (or some other white point), I was trying to think how that would work. The way we have been doing the built-in ACES configs, the non-D65 displays have a Bradford adaptation from the D65 of the reference space to their calibration white. 

I think we could use the D65 view transforms for a rendition that will be at the display's calibration white point but if we want to provide a true D65 or D60 version, we would need new view transforms. Those view transforms may need to undo the Bradford adaptation in the display.

Reply:
It seems to me the concept of "white point simulation" does not map easily to OCIO v2 viewing transforms workflows. I'm not sure if there is an ideal solution with the current system. As a further option, maybe one could also create a dedicated display for the D60 Sim in D65 case where the display side processing would chromatically adapt from D65 to D60, then apply a white scaling factor like is done in the builtin currently. This is assuming the builtin always "correctly" apply a chromatic adaptation to D65 as intended by cie_xyz_d65. This should allow to avoid duplicated view transforms?